src/wallet/wallet.h

```diff
@@ -700,7 +700,16 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
 
     bool fFileBacked;
 
-    std::set<int64_t> setKeyPool;
+    // class for holding a keypool-key-entry in memory
+    class KeypoolCache {
+    public:
+        int64_t nIndex;
+        CKeyID keyId;
```

Comment:
Note: `keyId` member is not actually used anywhere in this PR. (Fine, just pointing out to help review).